core/src/main/java/com/linecorp/armeria/server/RouteBuilder.java

```diff
@@ -0,0 +1,351 @@
+     * the {@code "userId"} path parameter.
+     */
+    public RouteBuilder regex(String regex) {
+        return regex(Pattern.compile(requireNonNull(regex, "regex")));
+    }
+
+    /**
+     * Sets the specified {@link Pattern} of the {@link Route}. The mapped {@link Service} is found when a
+     * {@linkplain ServiceRequestContext#path() path} matches the specified {@link Pattern}. It also extracts
+     * the values of the named groups to {@linkplain ServiceRequestContext#pathParams() pathParams}:
+     * e.g. {@code "^/users/(?<userId>[0-9]+)$"} will extract the second numeric part of the path into
+     * the {@code "userId"} path parameter.
+     */
+    public RouteBuilder regex(Pattern regex) {
+        return pathMapping(new RegexPathMapping(regex));
+    }
+
+    /**
+     * Sets the specified {@code prefix} and {@code pathPattern} of the {@link Route}. The mapped
+     * {@link Service} is found when a {@linkplain ServiceRequestContext#path() path} is under
+     * the specified {@code prefix} and the rest of the path matches the specified {@code pathPattern}.
+     *
+     * @see #path(String)
+     */
+    public RouteBuilder pathWithPrefix(String prefix, String pathPattern) {
+        prefix = ensureAbsolutePath(prefix, "prefix");
+        if (!prefix.endsWith("/")) {
+            prefix += '/';
+        }
+
+        if ("/".equals(prefix)) {
+            // pathPrefix is not specified or "/".
+            return path(pathPattern);
+        }
+
+        if (pathPattern.startsWith("/")) {
+            return path(concatPaths(prefix, pathPattern));
+        }
+
+        if (pathPattern.startsWith(EXACT)) {
+            return exact(concatPaths(prefix, pathPattern.substring(EXACT.length())));
+        }
+
+        if (pathPattern.startsWith(PREFIX)) {
+            return prefix(concatPaths(prefix, pathPattern.substring(PREFIX.length())));
+        }
+
+        if (pathPattern.startsWith(GLOB)) {
+            final String glob = pathPattern.substring(GLOB.length());
+            if (glob.startsWith("/")) {
+                return glob(concatPaths(prefix, glob));
+            }
+        }
+
+        return pathMapping(new PathMappingWithPrefix(prefix, getPathMapping(pathPattern)));
+    }
+
+    /**
+     * Sets {@link HttpMethod}s that this {@link Route} supports. If not set, the mapped {@link Service}
+     * accepts any {@link HttpMethod}s.
+     */
+    public RouteBuilder methods(HttpMethod... methods) {
+        methods(ImmutableList.copyOf(requireNonNull(methods, "methods")));
+        return this;
+    }
+
+    /**
+     * Sets {@link HttpMethod}s that this {@link Route} supports. If not set, the mapped {@link Service}
+     * accepts any {@link HttpMethod}s.
+     */
+    public RouteBuilder methods(Iterable<HttpMethod> methods) {
+        this.methods = ImmutableSet.copyOf(Sets.immutableEnumSet(requireNonNull(methods, "methods")));
```

Comment:
Don't need `ImmutableSet.copyOf`

Reply:
Thanks! didn't notice that they are copied inside.